heartbeat/monitors/monitor.go

```diff
@@ -227,6 +261,15 @@ func (m *Monitor) Start() {
 	for _, t := range m.watchPollTasks {
 		t.Start()
 	}
+
+	teleStats.monitors.Inc()
+
+	if stats, ok := teleStats.protocols[m.name]; !ok {
+		panic(fmt.Sprintf("Unknown protocol for monitor stats: %s", m.name))
```

Comment:
Should we really panic if can't to the stats? Assuming someone adds a new protocol, forgets to update the list and we do a release I would expect that we log an error but not stop the application because of it.

On the other hand having a panic makes it easy to spot the problem early on.

Reply:
I tend to favor panic for "This should never happen" sorts of things.

The benefit is that if we add a new protocol we won't forget stats support. In this case, I think its OK, because I can't see how we could go through a dev/test/release cycle without seeing the panic.

Reply:
OTOH, this breaks with the mock "test" monitor. Maybe it should jus be a log message.

WDYT? The other approach would be to add "test " as a protocol (or going further, to inject the map of monitors, but that seems like overkill).